### PR TITLE
Export NixOS module to be used from other configs

### LINKED ISSUE #17: Devise a better story for how people can use their own config after initial installation
Right now the README takes takes you up to installing the PoC config found in this repo, but no further. Given that the laptop is already usable as a daily driver if you are willing to live with some of the missing features, we might want to give people a bit more assistance with customizing the config and turning it into an actual usable installation. (Experienced Nix users can probably fairly quickly figure out a solution for this part already, but it would probably still be useful to have some recommended methods described in this repo, especially for more novice users. (E.g. see #1))

## Users without an existing config

If someone does not already have a config they want to bring with them, a workflow for how to customize the one found in this repo might be useful. (E.g. just clone the repo, commit your modifications, and let `git pull` create merge commits. The history might be ugly but it's an easy solution.)

## Users with an existing config

Right now I just copy paste stuff between my personal config and this repo, but this is obviously not a very convenient approach.

I think adding a NixOS module to the repo should not be too difficult, but a nixpkgs version mismatch is still a potential issue. (Fortunately patching nixpkgs is only required for the installer.)

## Questions
- Do we want to cater to non-flake users?
  While not fully, I think adding a `default.nix` or something to the repo should not be too difficult. Though I won't be testing any non-flake use-cases actively.

### PR DESCRIPTION
**Pull request by kuruczgy**
Fixes  #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,12 +102,18 @@ Next, mount the root filesystem and the EFI partition. (By default the EFI parti
...
 ```console
-# nixos-install --root /mnt --no-channel-copy --no-root-password --flake $(readlink /x1e-nixos-config)#system
+# nixos-install --root /mnt --no-channel-copy --no-root-password --flake x1e-nixos-config#system
 ```
 
+Explanation:
+
+- `--no-channel-copy` disables copying channels. If you would like to use a non-flake config with channels (which I don't recommend), you should omit this option.
+- `--no-root-password` disables setting a root password. Omit this if you don't have any other way to get into the installed system and need a root password.
+- `--flake x1e-nixos-config#system` specifies the example flake based configuration to be used that comes with this repository. Note that if you have already [created your own NixOS configuration](#making-your-own-nixos-configuration) for the to-be-installed system, you can use that here. (Either by specifying `--flake /path/to/your/config#your-hostname` or by placing your non-flake config at `/mnt/etc/nixos/configuration.nix` and omitting this option. Importantly, `nixos-generate-config` is not expected to work.) You can always switch to your own configuration later if you use the example config.
```
Comment (kuruczgy) [inline README.md:115]: > You can always switch to your own configuration later if you use the example config.

This is not strictly true, as the example config disables channels. We should warn users here that if they want a non-flake config, they have to use it from the start.

**Comment by kuruczgy** [comment]
TODO:
- Test module from my own config
- Test install again

**Comment by kuruczgy** [comment]
Approved, ready for merge.

**Comment by korken89** [comment]
Thank you for this, it helped a lot in merging this with existing config!

**Pull request merged**